javalib/src/main/scala/java/lang/String.scala

```diff
@@ -1,42 +1,908 @@
 }
 
 object _String {
-  def valueOf(obj: Object): _String = ???
-  def valueOf(data: Array[Char]): _String = ???
-  def valueOf(data: Array[Char], offset: scala.Int, count: scala.Int): _String = ???
-  def valueOf(b: scala.Boolean): _String = ???
-  def valueOf(c: scala.Char): _String = ???
-  def valueOf(s: scala.Short): _String = ???
-  def valueOf(i: scala.Int): _String = ???
-  def valueOf(l: scala.Long): _String = ???
-  def valueOf(f: scala.Float): _String = ???
-  def valueOf(d: scala.Double): _String = ???
-  def format(format: String, args: scala.Array[Object]): _String = ???
+  final val CASE_INSENSITIVE_ORDER: Comparator[_String] =
+    new CaseInsensitiveComparator()
+  private final val ascii = new Array[Char](128)
+  private var DefaultCharset: Charset = _
+  private var lastCharset: Charset    = _
+
+  for (i <- ascii.indices) {
+    ascii(i) = i.toChar
+  }
+
+  object ConsolePrintStream {
+    private var charset = "ISO-8859-1"
+//      AccessController.doPrivileged(new PriviActions[String]("console.encoding", "ISO8859_1"))
+
+    if (!Charset.isSupported(charset)) {
+      charset = "ISO-8859-1"
+    }
+  }
+
+  class ConsolePrintStream(out: java.io.OutputStream)
+      extends java.io.PrintStream(out, true) {
+    def print(str: _String) {
+      try {
+        if (str == null) {
+          write("null".getBytes(ConsolePrintStream.charset))
+        } else {
+          write(str.getBytes(ConsolePrintStream.charset))
+        }
+      } catch {
+        case e: java.io.IOException =>
+          setError()
+      }
+    }
+  }
+
+  private class CaseInsensitiveComparator
+      extends Comparator[_String] with Serializable {
+    def compare(o1: _String, o2: _String): Int =
+      o1.compareToIgnoreCase(o2)
+  }
+
+  def copyValueOf(data: Array[Char], start: Int, length: Int): _String =
+    new _String(data, start, length)
+
+  def copyValueOf(data: Array[Char]): _String =
+    new _String(data, 0, data.length)
+
+  def valueOf(data: Array[Char]): _String = new _String(data)
+
+  def valueOf(data: Array[Char], start: Int, length: Int): _String =
+    new _String(data, start, length)
+
+  def valueOf(value: Char): _String = {
+    val s =
+      if (value < 128) new _String(value, 1, ascii)
+      else new _String(0, 1, Array(value))
+    s.hashcode = value
+    s
+  }
+
+  def valueOf(value: scala.Double): _String = Double.toString(value)
+
+  def valueOf(value: scala.Float): _String = Float.toString(value)
+
+  def valueOf(value: scala.Int): _String = Integer.toString(value)
+
+  def valueOf(value: scala.Long): _String = Long.toString(value)
+
+  def valueOf(value: scala.Boolean): _String = if (value) "true" else "false"
+
+  def valueOf(value: AnyRef): _String =
+    if (value != null) valueOf(value.toString) else "null"
+
+  def format(_format: _String, args: AnyRef*): _String =
+    format(Locale.getDefault(), _format, args)
+
+  def format(loc: Locale, _format: _String, args: AnyRef*): _String = {
```

Comment:
This signature will not be binary compatible. In Java, varargs are encoded as arrays. The proper binary-compatible signature is (unfortunately) `args: Array[AnyRef]` (and no, not `Array[Any]`, as that erases to `Object` instead of `Array[Object]`)